server/src/test/java/org/elasticsearch/discovery/ClusterDisruptionCleanSettingsIT.java

```diff
@@ -0,0 +1,113 @@
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.indices.store.IndicesStoreIntegrationIT;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.test.disruption.NetworkDisruption;
+import org.elasticsearch.test.transport.MockTransportService;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, transportClientRatio = 0)
+public class ClusterDisruptionCleanSettingsIT extends ESIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(MockTransportService.TestPlugin.class);
+    }
+
+    /**
+     * This test creates a scenario where a primary shard (0 replicas) relocates and is in POST_RECOVERY on the target
+     * node but already deleted on the source node. Search request should still work.
+     */
+    public void testSearchWithRelocationAndSlowClusterStateProcessing() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        final String node_1 = internalCluster().startDataOnlyNode();
+
+        logger.info("--> creating index [test] with one shard and on replica");
+        assertAcked(prepareCreate("test").setSettings(
+            Settings.builder().put(indexSettings())
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0))
+        );
+        ensureGreen("test");
+
+        final String node_2 = internalCluster().startDataOnlyNode();
+        List<IndexRequestBuilder> indexRequestBuilderList = new ArrayList<>();
+        for (int i = 0; i < 100; i++) {
+            indexRequestBuilderList.add(client().prepareIndex().setIndex("test").setType("_doc")
+                .setSource("{\"int_field\":1}", XContentType.JSON));
+        }
+        indexRandom(true, indexRequestBuilderList);
+
+        IndicesStoreIntegrationIT.relocateAndBlockCompletion(logger, "test", 0, node_1, node_2);
+        // now search for the documents and see if we get a reply
+        assertThat(client().prepareSearch().setSize(0).get().getHits().getTotalHits(), equalTo(100L));
+    }
+
+    /**
+     * Tests that indices are properly deleted even if there is a master transition in between.
+     * Test for https://github.com/elastic/elasticsearch/issues/11665
+     */
+    public void testIndicesDeleted() throws Exception {
+        final Settings settings = Settings.builder()
+            .put(AbstractDisruptionTestCase.DEFAULT_SETTINGS)
+            .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "0s") // don't wait on isolated data node
+            .put(DiscoverySettings.COMMIT_TIMEOUT_SETTING.getKey(), "30s") // wait till cluster state is committed
```

Comment:
Because this adds to `AbstractDisruptionTestCase.DEFAULT_SETTINGS` I think it can remain in `ClusterDisruptionIT` and start up nodes with something like the following.

```
        final Settings settings = Settings.builder()
             .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "0s") // don't wait on isolated data node
             .put(DiscoverySettings.COMMIT_TIMEOUT_SETTING.getKey(), "30s") // wait till cluster state is committed
             .build();
        startMasterOnlyNodes(2, settings);
```

This is because the nodes start up with the settings from the test suite _plus_ those passed in explicitly:

https://github.com/elastic/elasticsearch/blob/185c06bb7f2bd7bff491738b01ed7716c69a70db/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java#L406-L423

Reply:
Nevermind, I misspoke :) will fix